### PR TITLE
Add scuffed battle.

### PR DESCRIPTION
**Pull request by GanyusLeftHorn**

### CODE REVIEW

**Review by memetrollsXD — COMMENTED** [review]
From just testing the command itself, it works on specific conditions only:

Tested using `battle 102202001 102202001`
- Plane ID needs to be correct type - Testing the above command on Forgotten Hall will brick the client, but if tested in Southern Snow Plains it will work perfectly.

File: src/commands/battle.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                    equipmentList: equipData,
+                    hp: avatar.db.fightProps.hp,
+                    id: avatar.db.baseAvatarId,
+                    index: i,
+                    level: avatar.db.level,
+                    promotion: avatar.db.promotion,
+                    rank: avatar.db.rank,
+                    relicList: [],
+                    skilltreeList: []
+                } as unknown as BattleAvatar;
```
Comment (TheLostTree) [inline src/commands/battle.ts:59]: try to use BattleAvatar.fromPartial instead of the as keyword when possible

File: src/commands/battle.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            }),
+            buffList: [] as BattleBuff[],
+            battleId: 0,
+            heroPathList: [] as HeroPath[],
```
Comment (TheLostTree) [inline src/commands/battle.ts:63]: y'all can do ``new Array<HeroPath>() ``as well

**Review by TheLostTree — CHANGES_REQUESTED** [review]
Generally I'd prefer if y'all didn't use the as keyword 😔 type.fromPartial() should be your goto unless it won't work otherwise

**Review by TheLostTree — APPROVED** [review]
so hot

**Pull request merged**